test/schema-serialization.test.js

```diff
@@ -60,6 +60,95 @@ test('custom serializer options', t => {
   })
 })
 
+test('Different content types', t => {
+  t.plan(15)
+
+  const fastify = Fastify()
+  fastify.addSchema({
+    $id: 'test',
+    type: 'object',
+    properties: {
+      image: { type: 'string' }
+    }
+  })
+
+  fastify.get('/', {
+    schema: {
+      response: {
+        200: [
+          { content: 'application/json', type: { id: { type: 'number' }, name: { type: 'string' } } },
+          {
+            content: 'application/vnd.v1+json',
+            type: {
+              type: 'array',
+              items: { $ref: 'test' }
+            }
+          }
+        ],
+        '3xx': [
+          { content: 'application/vnd.v2+json', type: { name: { type: 'string' } } }
+        ]
+      }
+    }
+  }, function (req, reply) {
+    switch (req.headers.accept) {
+      case 'application/json':
+        reply.header('Content-Type', 'application/json')
+        reply.send({ id: 1, name: 'Foo', image: 'BIG IMAGE' })
+        break
+      case 'application/vnd.v1+json':
+        reply.header('Content-Type', 'application/vnd.v1+json')
+        reply.send([{ id: 1, name: 'Foo', image: 'BIG IMAGE' }, { id: 2, name: 'Foo2', image: 'BIG IMAGE2' }])
+        break
+      case 'application/vnd.v2+json':
+        reply.header('Content-Type', 'application/vnd.v2+json')
+        reply.code(300)
+        reply.send({ id: 1, name: 'Foo', image: 'BIG IMAGE' })
```

Comment:
could you add more variation between the schema?
Such as additional fields etc..